scripts/irods/database_upgrade.py

```diff
@@ -117,6 +117,19 @@ def run_update(irods_config, cursor):
         database_connect.execute_sql_statement(cursor, "update R_TOKN_MAIN set token_name = 'read_object' where token_name = 'read object';")
         database_connect.execute_sql_statement(cursor, "update R_TOKN_MAIN set token_name = 'read_metadata' where token_name = 'read metadata';")
 
+    elif new_schema_version == 10:
+        if irods_config.catalog_database_type == 'mysql':
+            # Assume the database is MySQL 8 or later (support for the SQL WITH-clause was not added until MySQL 8).
+            if len(database_connect.execute_sql_statement(cursor, "select alias from R_SPECIFIC_QUERY where alias = 'DataObjInCollReCur';").fetchall()) == 0:
```

Comment:
there is also the syntax of `INSERT … ON DUPLICATE KEY UPDATE`

as an option.

this is fine as is - easily readable - two queries instead of one, but that's not a problem.

Reply:
I thought about searching for alternative syntax that makes this a one liner, but decided against it because it didn't seem worth it. And I doubt most people on the team would understand the syntax without having to google it (including myself).

Willing to change what I have already if others feel it's worth it.

Reply:
Not worth it to me.